src/Interpreters/PredicateExpressionsOptimizer.cpp

```diff
@@ -72,6 +84,9 @@ std::vector<ASTs> PredicateExpressionsOptimizer::extractTablesPredicates(const A
             return {};   /// Not optimized when predicate contains stateful function or indeterministic function or window functions
         }
 
+        if (hasInputTableFunction(predicate_expression))
+            return {}; /// Not optimized when predicate contains input table function
```

Comment:
```suggestion
            return {}; /// Not optimized when predicate contains input table function
```
I think this comment is not useful. From the if statement it is clear when we don't want to optimize. However it might worth to note _why_: `input table function can only be read once, but the optimization might copy the subquery which would result in reading more from StorageInput then it expects`. Or something along this line.